File: ios/GoogleNavigationSdk.m
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(GoogleNavigationSdk, NSObject)

RCT_EXTERN_METHOD(provideAPIKey:(NSString)apiKey
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup {
    return NO;
}

@end

File: ios/GoogleNavigationSdk.swift
import GoogleMaps
import GoogleNavigation

@objc(GoogleNavigationSdk)
class GoogleNavigationSdk: NSObject {
    
    @objc(provideAPIKey:withResolver:withRejecter:)
    func provideAPIKey(apiKey: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        GMSServices.provideAPIKey(apiKey)
        resolve()
    }
}
